addons/website/static/src/snippets/s_timeline/000.scss

```diff
@@ -59,6 +59,7 @@
         span:not(.fa) {
             display: inline-block;
             padding: 5px;
+            background: $body-bg;
```

Comment:
The purpose was to adapt the date element background to the page background, but indeed ... it doesn't work when a preset color is defined.
It would be nice if the date element background could match to the current background used (page or preset).
 
| Before | After |
|--------|--------|
| ![Capture d’écran 2023-10-05 à 15 50 33](https://github.com/odoo/odoo/assets/80679690/8366233c-245b-47a9-8db4-8087a945e436) | ![Capture d’écran 2023-10-05 à 15 48 24](https://github.com/odoo/odoo/assets/80679690/4fe4d196-82d1-4a7f-aa62-bf0ab0127d8a) |

Reply:
Mmh indeed. Can't think of a good solution given the current layout. Maybe `background-color: var(--o-cc-bg, var(--body-bg))` and setting `--o-cc-bg: XXX` with the right color for all `o_ccX` classes could work. What do you think?

Reply:
I tested before sending my message actually and it works, I was wondering if it was the right way to go though. In fact I am thinking about going all in and reviewing the way o_cc classes are defined. Basically instead of producing this kind of CSS:

```css
.o_cc1 {
    background-color: blue;
}
.o_cc1 a {
    color: green;
}
.o_cc2 {
    background-color: red;
}
.o_cc2 a {
    color: orange;
}
.o_cc3 {
    background-color: purple;
}
.o_cc3 a {
    color: pink;
}
```

I would go for (something like you did with `o-input-bg`):
```css
.o_cc {
    background-color: var(--o-cc-bg);
}
.o_cc a {
    color: var(--o-cc-a-color);
}
.o_cc1 {
    --class-bg: blue;
    --class-a-color: green;
}
.o_cc2 {
    --class-bg: red;
    --class-a-color: orange;
}
.o_cc3 {
    --class-bg: purple;
    --class-a-color: pink;
}
```

It obviously leads to less CSS (instead of `number_of_component_customized_per_cc x number_of_cc` it would be `number_of_component_customized_per_cc + number_of_cc` but I am wondering if it is a good practice (less CSS may not be more efficient or more readable CSS)... any opinion? cc @stefanorigano

Reply:
I think that css variables have been designed exactly for this purpose :)

Some (very general) notes:

- quite obvious, but once a color has been "printed" as a CSS variable it's quite difficult to interpolate. We solved the issue with color-mix but a fallback has to be defined each time, and it would not be always feasible in other circumstances.

- in this PR we have set $border and BS input variable to CSS variables, but for other elements it's not always possible: sometimes BS tries to compute these values and fails.

Anyways, BS is going in this direction (see the new [.btn](https://github.com/twbs/bootstrap/blob/main/scss/_buttons.scss) class)... maybe we should wait to migrate to these versions before reworking the entire color system ?

Reply:
> maybe we should wait to migrate to these versions before reworking the entire color system ?

I guess, still would need this to solve the timeline stuff but I'll do it only for the background with a comment for the future then :+1: